src/controllers/py.controller.ts

```diff
@@ -0,0 +1,163 @@
+    private static readonly _description: string = 'Local python execution';
+    private static readonly _controllerId: string = 'py-local';
+    private static readonly _notebookType: string = 'pythonbook';
+    private static readonly _label: string = 'Python Local';
+    private static readonly _execMeta: string = 'execution';
+    private static readonly _defaultExecutionCmd: string = `python`;
+    private static readonly _execFileArg: string = 'xprog';
+
+    constructor() {
+        super(
+            PyController._controllerId,
+            PyController._notebookType,
+            PyController._label,
+        );
+    }
+    cellMetadata(): MetadataField[] | undefined {
+        return [
+            {
+                key: PyController._execMeta,
+                default: PyController._defaultExecutionCmd,
+                description: `where {${PyController._execFileArg}} is the path to the temporary file to execute`
+            }
+        ];
+    }
+    notebookMetadata(): MetadataField[] | undefined {
+        return;
+    }
+
+    public supportedLanguages(): string[] | undefined {
+        return PyController._supportedLanguages;
+    }
+
+    public supportsExecutionOrder(): boolean | undefined {
+        return false;
+    }
+
+    public detail(): string | undefined {
+        return PyController._detail;
+    }
+
+    public description(): string | undefined {
+        return PyController._description;
+    }
+
+    public contributors(): Contributor[] | undefined {
+        return [
+            {
+                name:  'Artsem Hutarau',
+                email: 'gutorov.artem@yandex.ru',
+                url:   'https://github.com/MonkeyBuisness'
+            },
+            {
+                name: 'Raman H',
+                email: [
+                    'e', 'm', '.', 'm',
+                    'p', '@', 'h', '.',
+                    'n', 'a', 'm', 'a',
+                    'r'
+                ].reverse().join(''),
+                url: 'https://github.com/duckyou',
+            },
+        ];
+    }
+
+    public icon(): string | undefined {
+        return 'py-local.png';
+    }
+
+    public gettingStartedGuide(): string | undefined {
+        return 'py-local.md';
+    }
+
+    public async execute(ex: NotebookCellExecution): Promise<boolean | undefined> {
+        let execCmd: string = (ex.cell.metadata || {})[PyController._execMeta];
+        if (!execCmd) {
+            execCmd = PyController._defaultExecutionCmd;
+        }
+
+        // save to tmp file
+        const tmpFileName = `notebook.${uuid4()}.py`;
+        const tmpFile = await UniqueFileName.get(path.join(tmpdir(), tmpFileName));
+        fs.writeFileSync(tmpFile, ex.cell.content);
+        execCmd = [execCmd, tmpFile].join(' ');
+
+        // exec command
+        let success: boolean | undefined;
+        const commands = execCmd.split(' ');
+        let process: any = undefined;
+        try {
```

Comment:
It would be nice, if you start using a [new way](https://github.com/MonkeyBuisness/cellementary-extension/blob/master/src/utils/executor.util.ts) to execute commands.

For instance, this is the execution implementation for the `go-local` notebook:
https://github.com/MonkeyBuisness/cellementary-extension/blob/055c2e49532c7443ca8c32dd0cf8fb1ac6a0ee18/src/controllers/go.controller.ts#L78-L110